arcgis-ios-sdk-samples/Utility Network/Trace utility network/TraceUtilityNetworkViewController.swift

```diff
@@ -316,12 +351,35 @@ class ConnectedTraceViewController: UIViewController, AGSGeoViewTouchDelegate {
         }
     }
     
+    // MARK: Set trace type
+    @IBAction func setTraceType(_ sender: Any) {
+        let alertController = UIAlertController(title: "Select trace type", message: nil, preferredStyle: .actionSheet)
+        let types = [
+            ("Connected", AGSUtilityTraceType.connected),
+            ("Subnetwork", AGSUtilityTraceType.subnetwork),
+            ("Upstream", AGSUtilityTraceType.upstream),
+            ("Downstream", AGSUtilityTraceType.downstream)
+        ]
+        types.forEach { type in
+            let action = UIAlertAction(title: type.0, style: .default) { [unowned self] _ in
+                self.traceType = type
+                self.setStatus(message: "Trace type \(type.0) selected.")
+            }
+            alertController.addAction(action)
+        }
```

Comment:
Referencing to tuple members without labels really harms readability. It would be helpful to add labels to the tuple members:

```suggestion
        let types: [(name: String, type: AGSUtilityTraceType)] = [
            ("Connected", .connected),
            ("Subnetwork", .subnetwork),
            ("Upstream", .upstream),
            ("Downstream", .downstream)
        ]
        types.forEach { (name, type) in
            let action = UIAlertAction(title: name, style: .default) { [unowned self] _ in
                self.traceType = type
                self.setStatus(message: "Trace type \(name) selected.")
            }
            alertController.addAction(action)
        }
```

This suggestion requires a couple other changes. The type of `traceType` would need to be `AGSUtilityTraceType`, which I don't think will cause any problems.

Reply:
I keep the name string of the trace type so that it can be displayed in the progress HUD with correct type name. Previously it was hardcoded as "Connected" here: https://github.com/Esri/arcgis-runtime-samples-ios/blob/c86a0f2155c7e84d716fb9c5c24fdcb8629fced2/arcgis-ios-sdk-samples/Utility%20Network/Connected%20Trace/ConnectedTraceViewController.swift#L191